### PR TITLE
rename `LeftNavigation` to `LeftNav` (and associated touchups)

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Left Nav](https://deploy-preview-34--neo-react-library-storybook.netlify.app/?path=/story/components-left-navigation)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

Removed `Logo` and `LogoLink` and replaced with `Image` and `ImageLink`. Added `isDecorativeOrBranding` prop to `Image` so that we can better check for a11y issues in that component. 

Renamed `LeftNavigation` to `LeftNav`. This rename is the vast majority of changes due to file+folder renames. Will be updating `Navbar` to `TopNav` in next PR. Updated `LeftNav` sub-components to use `LeftNav.*` in order to remove confusion, see end of file `LeftNav.tsx` and it's props file for implementation details. Note that internally we can still use the subcomponents (for testing and such), but to be used externally they _must_ be used like `LeftNav.*`

Centralized Left Nav stories. Again, didn't really change anything, just moved them around. 

Commented out some tests from `TopLinkItem.test.jsx` as they didn't make sense to me. If you know what they're supposed to do, please open a PR after this one fixing them.

### CODE REVIEW

File: src/components/Image/Image.tsx [hunk 1]
```diff
@@ -10,33 +10,38 @@ import {
...
+  isDecorativeOrBranding = false,
   onError,
   onLoad,
   src,
   style,
   thumbnail = false,
   ...rest
 }: ImageProps) => {
-  if (!alt) {
+  if (!alt && !isDecorativeOrBranding) {
```
Comment (enrique-prado) [inline src/components/Image/Image.tsx:44]: Nice

File: src/components/Navbar/index.ts [hunk 2]
```diff
@@ -1,3 +1,2 @@
 export * from "./Navbar";
-export * from "./LeftContent";
```
Comment (enrique-prado) [inline src/components/Navbar/index.ts:2]: Did you mean to remove this?

**Review by enrique-prado — APPROVED** [review]
Left a question.

File: src/components/Navbar/index.ts (hunk 2 above)
Comment (joe-s-avaya) [reply src/components/Navbar/index.ts:2]: Yes, `LeftContent` only had `Logo` components in it, which have been removed and replaced with the `Image` components.

File: src/components/ImageLink/ImageLink.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * A `ImageLink` is an image that is wrapper in an anchor tag
+ *
+ * @example
+ * <ImageLink href="www.homepage.com" src="link/to/image.png" alt="Image description" />
+ * <ImageLink href="www.homepage.com" src="link/to/image.png" alt="Image description" target="_blank" />
+ */
+export const ImageLink = ({
+  alt,
```
Comment (yangAtSpoken) [inline src/components/ImageLink/ImageLink.tsx:22]: It seems "alt" should be a required prop.  Is it so?

File: src/components/LeftNav/TopLinkItem/TopLinkItem.test.jsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      <ul>
+        <LeftNav.TopLinkItem label={TopLinkItemLabel} />
+      </ul>
+    );
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
+
+  // TODO: skipping due to not seeing an `active` prop being used
+  // it("assigns the appropriate class name when the `active` prop is passed", () => {
```
Comment (yangAtSpoken) [inline src/components/LeftNav/TopLinkItem/TopLinkItem.test.jsx:31]: Is coverage down with these commented out?

**Review by yangAtSpoken — COMMENTED** [review]
some comments

**Review by yangAtSpoken — APPROVED** [review]

File: src/components/LeftNav/TopLinkItem/TopLinkItem.test.jsx (hunk 4 above)
Comment (joe-s-avaya) [reply src/components/LeftNav/TopLinkItem/TopLinkItem.test.jsx:31]: nope, strangely

File: src/components/ImageLink/ImageLink.tsx (hunk 3 above)
Comment (joe-s-avaya) [reply src/components/ImageLink/ImageLink.tsx:22]: It is required yes. `ImageLInkProps` extends `ImageProps`, which has logic that either forces `alt` to be defined, or to have `isDecorativeOrBranding` defined.

**Pull request merged**